koiki/woocommerce/state.py

```diff
@@ -0,0 +1,27 @@
+import pycountry
+
+
+class InvalidState(Exception):
+    pass
+
+
+class State():
+    def __init__(self, code):
+        full_code = f'ES-{code}'
+
+        if code == 'Barcelona':
+            full_code = 'ES-B'
+        elif code == 'Girona':
+            full_code = 'ES-GI'
+        elif code == 'Tarragona':
+            full_code = 'ES-T'
+        elif code == 'Lleida':
+            full_code = 'ES-L'
```

Comment:
IMO we don't need any more abstract approach than this, which is pretty explicit. Other solutions that pretend to be clever may cause more trouble now.

Reply:
confirmed by @sseerrggii. La Zona's scope will be Catalonia only.